pkg/local_object_storage/metabase/exists.go

```diff
@@ -91,21 +92,32 @@ func objectStatus(tx *bbolt.Tx, addr oid.Address, currEpoch uint64) uint8 {
 	// GC is expected to collect all the objects that have
 	// expired previously for less than the one epoch duration
 
-	var expired bool
-	oID := addr.Object()
-	cID := addr.Container()
-
-	// bucket with objects that have expiration attr
-	attrKey := make([]byte, bucketKeySize+len(objectSDK.AttributeExpirationEpoch))
-	expirationBucket := tx.Bucket(attributeBucketName(cID, objectSDK.AttributeExpirationEpoch, attrKey))
-	if expirationBucket != nil {
-		// bucket that contains objects that expire in the current epoch
-		prevEpochBkt := expirationBucket.Bucket([]byte(strconv.FormatUint(currEpoch-1, 10)))
-		if prevEpochBkt != nil {
-			rawOID := objectKey(oID, make([]byte, objectKeySize))
-			if prevEpochBkt.Get(rawOID) != nil {
-				expired = true
-			}
+	var (
+		expired        bool
+		oID            = addr.Object()
+		cID            = addr.Container()
+		metaBucketName = append([]byte{metadataPrefix}, cID[:]...)
+		metaBucket     = tx.Bucket(metaBucketName)
+	)
+
+	if metaBucket != nil {
+		var (
+			cur       = metaBucket.Cursor()
+			expPrefix = make([]byte, attrIDFixedLen+len(objectSDK.AttributeExpirationEpoch))
+		)
+		expPrefix[0] = metaPrefixIDAttr
+		copy(expPrefix[1:], oID[:])
+		copy(expPrefix[1+len(oID):], objectSDK.AttributeExpirationEpoch)
+
+		expKey, _ := cur.Seek(expPrefix)
+		if bytes.HasPrefix(expKey, expPrefix) {
+			var val = expKey[len(expPrefix):]
```

Comment:
`+attributeDelimiterLen`?

Reply:
It's already included (`attrIDFixedLen`), just not copied since it's zero.

Reply:
tricky for me, it's worth leaving a comment at least

Reply:
excuse me?.. we have a separate var for it, use it everywhere, and you decided not to use it only here because it is the same as the default value for a byte? just imagine we decide to change it to smth else once in the future and this code is still the same

Reply:
see comment now, it's enough

Reply:
> It can't realistically be changed to me

https://github.com/nspcc-dev/neofs-node/pull/3168? Am I right that the argument is like "I think we will never change code/make a mistake"?

Reply:
> see comment now, it's enough

when we change the delimiter, we change const's value, we do not read every comment in every metabase's file

Reply:
It _can not_ be changed. Any other value breaks the basic properties of these indexes. That's exactly why #3168 fixed it from the value that just doesn't really work (but was brought in because 0 is a valid UTF-8).